daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -790,53 +789,47 @@ private[lf] final class Compiler(
     //       <r> = $catch ([update] <token>) true false
     //   in $endCommit(mustFail = true) <r> <token>
     withEnv { _ =>
-      withLabel(
-        "submitMustFail",
-        SEAbs(1) {
-          val tokenPos = nextPosition()
-          val beginCommit = SBSBeginCommit(optLoc)(compile(party), svar(tokenPos))
-          nextPosition()
-          val result =
-            SECatch(SEApp(compile(update), Array(svar(tokenPos))), SEValue.True, SEValue.False)
-          val resultPos = nextPosition()
-
-          SELet(
-            beginCommit,
-            result,
-          ) in SBSEndCommit(mustFail = true)(svar(resultPos), svar(tokenPos))
+      labeledFunction("submitMustFail") { tokenPos =>
+        let(SBSBeginCommit(optLoc)(compile(party), svar(tokenPos))) { _ =>
+          let(SECatch(app(compile(update), svar(tokenPos)), SEValue.True, SEValue.False)) {
+            resultPos =>
+              SBSEndCommit(mustFail = true)(svar(resultPos), svar(tokenPos))
+          }
         }
-      )
+      }
     }
 
   @inline
   private[this] def compileGetParty(expr: Expr): SExpr =
-    function(1, "getPArty") {
-      case List(tokenPos) => SBSGetParty(compile(expr), svar(tokenPos))
+    labeledFunction("getPArty") { tokenPos =>
```

Comment:
`"getPArty"` with a capital `A` !! (I realise it was like this before)

Reply:
good catch!